pymc/model.py

```diff
@@ -1765,8 +1765,8 @@ def set_data(new_data, model=None):
     model = modelcontext(model)
 
     for variable_name, new_value in new_data.items():
-        model.set_data(variable_name, new_value)
-
+        model.set_data(variable_name, new_value,coords)
```

Comment:
```suggestion
        model.set_data(variable_name, new_value, coords=coords)
```

Here I think it would be best to use the argument as keyword instead of positional, but that should not change the output.